src/PowerShellEditorServices/Services/PowerShell/Utility/RunspaceExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.PowerShell.EditorServices.Utility;
+using System;
+using System.Linq.Expressions;
+using System.Management.Automation;
+using System.Reflection;
+using System.Threading;
+
+namespace Microsoft.PowerShell.EditorServices.Services.PowerShell.Utility
+{
+    using System.Management.Automation.Runspaces;
+
+    internal static class RunspaceExtensions
+    {
+        private static readonly Action<Runspace, ApartmentState> s_runspaceApartmentStateSetter;
+
+        private static readonly Func<Runspace, string, string> s_getRemotePromptFunc;
+
+        static RunspaceExtensions()
+        {
+            // PowerShell ApartmentState APIs aren't available in PSStandard, so we need to use reflection.
+            if (!VersionUtils.IsNetCore || VersionUtils.IsPS7OrGreater)
```

Comment:
We'll want to remove the whole if statement because we were attempting to run separate code for PowerShell 6 where a hole in the support existed, but that support has been discontinued.